crates/autopilot/src/solvable_orders.rs

```diff
@@ -807,20 +808,13 @@ mod tests {
         );
         let metrics = Metrics::instance(observe::metrics::get_storage_registry()).unwrap();
 
-        // We'll have no native prices in this call. But this call will cause a
-        // background task to fetch the missing prices so we'll have them in the
-        // next call.
-        let (filtered_orders, prices) =
-            get_orders_with_native_prices(orders.clone(), &native_price_estimator, metrics);
-        assert!(filtered_orders.is_empty());
-        assert!(prices.is_empty());
-
-        // Wait for native prices to get fetched.
-        tokio::time::sleep(tokio::time::Duration::from_millis(10)).await;
-
-        // Now we have all the native prices we want.
-        let (filtered_orders, prices) =
-            get_orders_with_native_prices(orders.clone(), &native_price_estimator, metrics);
+        let (filtered_orders, prices) = get_orders_with_native_prices(
+            orders.clone(),
```

Comment:
This is very expensive. Consider passing `orders: &mut [Order]`

Reply:
This is a unit test with 4 orders... 
Also adjusting the function would make the code slower in the real world use case since `&mut [Order]` can only be reordered but not filtered.

Reply:
> This is a unit test with 4 orders...

My bad, I don't know how I overlooked that :/ Got it!